### PR TITLE
Adding ability to use private registries

### PR DESCRIPTION
**Pull request by mkubarycz**

### CODE REVIEW

**Comment by seiflotfy** [comment]
testing it in a bit

**Comment by treeder** [comment]
This looks great!  But looks like there may be some overlap with #547, which adds an `fn docker login` command. I have a feeling these can coexist as yours appears to take env vars whereas #547 stores the auth in the database (which may make more sense when we have support for authenticating different users and they use their own registry credentials). 

Also, could you add some docs for this?

**Comment by seiflotfy** [comment]
@mkubarycz could you add some docs?

File: api/runner/task.go [hunk 1]
```diff
@@ -2,10 +2,12 @@ package runner
```
Comment (denismakogon) [inline api/runner/task.go]: Shouldn't this code be at iron-io/runner rather than in functions? All this code is driver-specific.

**Comment by pedronasser** [comment]
@mkubarycz I've taken the freedom of sending a PR to your fork, adding something. Please check it.
Beside that IMO this PR looks good.

I agree the ideal state would have been adding this to the iron/runner. But since we currently only supporting Docker and the code related to it is quite modular, in the future we can separate it from this repo.

I think this PR is a start towards supporting private registries and this can be improved in the future

cc @treeder @seiflotfy

**Comment by pedronasser** [comment]
Making a new PR to continue this PR

**Pull request closed without merging**